### PR TITLE
feat(delete): add after_delete config for post-delete YAML cleanup

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description

- Link to the related issue(s): Issue #994
- Adds a `deleting.after_delete` config toggle that controls what happens to the local ad YAML file after the standalone `delete` command runs. Three modes: `NONE` (default, no change), `RESET` (clear server-side metadata so the ad is eligible for republication as "new"), `DISABLE` (set `active: false` so the ad is skipped until manually re-enabled).

## 📋 Changes Summary

- Added `DeletingConfig` model with `after_delete: Literal["NONE", "RESET", "DISABLE"]` field (default `"NONE"`)
- Added `deleting: DeletingConfig` field to `Config` class
- Added `_apply_after_delete_policy()` helper that mutates `Ad` model and dict in-place based on mode
- Integrated policy in `delete_ads` with detection heuristic: cleanup runs when `delete_ad` returns `True` (200) or when `ad_cfg.id` transitions from non-None to None (404)
- Policy does NOT apply during publish-time `delete_ad` call sites — only the standalone delete command
- Regenerated `docs/config.default.yaml` and `schemas/config.schema.json`

### ⚙️ Type of Change
Select the type(s) of change(s) included in this pull request:
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (adds new functionality without breaking existing usage)
- [ ] 💥 Breaking change (changes that might break existing user setups, scripts, or configurations)

## ✅ Checklist
Before requesting a review, confirm the following:
- [x] I have reviewed my changes to ensure they meet the project's standards.
- [x] I have tested my changes and ensured that all tests pass  (`pdm run test`).
- [x] I have formatted the code (`pdm run format`).
- [x] I have verified that linting passes (`pdm run lint`).
- [x] I have updated documentation where necessary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

## Summary by CodeRabbit

* **New Features**
  * Added configurable post-delete cleanup options to control how local ad records are handled after deletion (leave, reset metadata, or disable).

* **Bug Fixes**
  * More accurate detection of attempted vs. successful deletions and correct incrementing of deletion counts.
  * Ensure metadata cleanup is skipped when no targets are found while still handling 404-style removals appropriately.

* **Tests**
  * Added unit tests covering post-delete behaviours and edge cases.

### CODE REVIEW

File: src/kleinanzeigen_bot/__init__.py [hunk 1]
```diff
@@ -183,6 +183,50 @@ def _relative_ad_path(ad_file:str, config_file_path:str) -> str:
...
+        ad_cfg.repost_count = 0
+        ad_cfg.price_reduction_count = 0
+        return True
+
+    if mode == "DISABLE":
+        ad_cfg.active = False
+        ad_cfg_orig["active"] = False
+        return True
+
+    return False
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/__init__.py:238]: Dismissed. The six fields are a few lines apart in the same private function with a single call site. Introducing a mapping dict to handle two different semantics (dict-key removal vs model-field assignment with type-specific defaults where some reset to `None` and others to `0`) adds indirection for near-zero drift protection on a rarely-changed function. The explicit assignments also serve as documentation of what each field resets to — a `setattr` loop would obscure that. Not worth the abstraction here.

File: tests/unit/test_config_model.py [hunk 2]
```diff
@@ -404,3 +404,21 @@ def test_diagnostics_legacy_capture_migration(
...
+@pytest.mark.parametrize("value", ["NONE", "RESET", "DISABLE"])
+def test_deleting_config_accepts_valid_values(minimal_config:dict[str, object], value:str) -> None:
+    cfg = {**minimal_config, "deleting": {"after_delete": value}}
+    config = Config.model_validate(cfg)
+    assert config.deleting.after_delete == value
+
+
+def test_deleting_config_rejects_invalid_value(minimal_config:dict[str, object]) -> None:
+    cfg = {**minimal_config, "deleting": {"after_delete": "INVALID"}}
+    with pytest.raises(Exception, match = "after_delete"):
```
[the remark this answers is not in the document]
Comment (1cu) [reply tests/unit/test_config_model.py:423]: Dismissed — the test file consistently uses `Exception` as the catch type for validation errors throughout (see the surrounding test methods in the same class). The project wraps Pydantic `ValidationError` in a custom `ContextualValidationError`, so coupling tests to `ValidationError` would be fragile. The generic `Exception` pattern matches the rest of the file and is intentional.

File: tests/unit/test_init.py [hunk 3]
```diff
@@ -2891,88 +2905,189 @@ class TestKleinanzeigenBotAdDeletion:
...
+        ):
+            result = await test_bot.delete_ad(ad_cfg, published_ads, delete_old_ads_by_title = True)
 
-        published_ads = [{"title": "Different Title", "id": "12345"}, {"title": "Other Title", "id": "11111"}]
+        assert result is False
+        assert ad_cfg.id == 99999  # Preserved — no deletion attempted
+        mock_web_open.assert_not_called()
+        mock_web_find.assert_not_called()
+        mock_web_sleep.assert_not_called()
+        mock_request.assert_not_called()
```
[the remark this answers is not in the document]
Comment (1cu) [reply tests/unit/test_init.py:2964]: Dismissed — the two test cases cover different scenarios: one tests the no-match early-return path (no ad found by selector, so no delete attempted), the other tests deduplication logic within a matched ad. They are not asserting contradictory contracts. This test structure is pre-existing from PR #995 and is not changed by this PR.

File: tests/unit/test_init.py [hunk 4]
```diff
@@ -5081,3 +5196,98 @@ async def test_extract_ad_id_falls_back_to_script_when_referrer_lacks_confirmati
...
+        if expect_metadata_cleared:
+            assert ad_cfg.id is None
+            assert ad_cfg.repost_count == 0
+            assert "id" not in ad_cfg_orig
+            mock_save.assert_called_once()
+        else:
+            assert "id" in ad_cfg_orig
+            assert ad_cfg.repost_count == 3
+            if mode == "NONE":
+                mock_save.assert_not_called()
```
[the remark this answers is not in the document]
Comment (1cu) [reply tests/unit/test_init.py:5329]: Fixed in commit 3a9d4ba. Both RESET and DISABLE modes now assert `mock_save.assert_called_once()`, and NONE asserts `mock_save.assert_not_called()`. The RESET branch also asserts full contract: `ad_cfg.id is None`, `ad_cfg.repost_count == 0`, `"id" not in ad_cfg_orig`, plus the dedicated `test_cleanup_on_404_detection` test verifies the 404 detection path with its own save assertion.
